src/hooks/useKeyDown.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export default function useKeyDown(targetKey: string, callbackFn: () => void) {
+  // State for keeping track of whether key is pressed
+  const [keyPressed, setKeyPressed] = useState<boolean>(false);
+
+  // If pressed key is our target key then set to true
+  const pressHandler = useCallback(
+    ({ key }: KeyboardEvent) => {
+      if (key === targetKey) {
+        setKeyPressed(true);
+      }
+    },
+    [targetKey]
+  );
+
+  // Add event listeners
+  useEffect(() => {
+    window.addEventListener('keyup', pressHandler);
+    return () => {
+      window.removeEventListener('keyup', pressHandler);
+    };
+  }, [pressHandler]);
+
+  // useEffect ensures that the callback function is called once, which is relevant when navigating
+  useEffect(() => {
+    // If the user is not currently focused on the body, return
+    const activeEl = document.activeElement;
+    if (activeEl?.tagName !== 'BODY') return;
+
+    if (keyPressed) {
+      callbackFn();
+      // After executing, reset keyPressed to false so that the user can click the same key again on the same page
+      // Only currently relevant if user is on /edit -> escapes -> shows modal -> escapes -> clicks escape again
+      // This also prevents route-based navigation from duplicating
+      setKeyPressed(false);
+    }
+  }, [keyPressed, callbackFn]);
+}
```

Comment:
@Robinnnnn I fixed the error we were talking about at the end of our call by setting `setKeyPressed` to false right after execution of the callback function. 

This allows us to include `callbackFn` in the dependency array now (so maybe I was wrong about the reasons for the loop, but this fixes 👍)

Reply:
NIC!!!